test/s3_unit_tests.cc

```diff
@@ -419,22 +419,22 @@ TEST_F(FileSystemS3Fixture, StatRoot) {
 	struct stat buf;
 	ASSERT_EQ(fs.Stat("/test", &buf, 0, nullptr), 0);
 
-	ASSERT_EQ(buf.st_mode & S_IFDIR, S_IFDIR);
+	ASSERT_EQ(static_cast<int>(buf.st_mode & S_IFDIR), S_IFDIR);
```

Comment:
If we need to cast, I think it'd be better to keep everything as `mode_t` as in:

```suggestion
	ASSERT_EQ(buf.st_mode & S_IFDIR, static_cast<mode_t>(S_IFDIR));
```

However, I haven't noticed this comparison issue.  What platform are you hitting this on?